src/pg/mod.rs

```diff
@@ -724,6 +730,21 @@ impl AsyncPgConnection {
             .unwrap_or_else(|p| p.into_inner())
             .on_connection_event(event);
     }
+
+    pub fn notification_stream(
```

Comment:
So you mentioned the behavior for not having a Connection should be having a `pending stream`. The way I did makes it so the stream is just `empty`. I think the empty (besides being easier to code) makes more sense because we could never receive a notification and will point the user to the documentation explaining why it's empty. What do you think?

Reply:
Got it. The only way I could achieve this behavior was through a manual stream implementation, do you know of a different way? Example: 910738116c30835d4571ea0170d25342c3598bc5

Reply:
Maybe just return a `futures::either::Either` stream instead? https://docs.rs/futures/0.3.31/futures/future/enum.Either.html. We could use the channel as one variant and a `futures::stream::pending` as the other variant (https://docs.rs/futures/0.3.31/futures/prelude/stream/fn.pending.html)

(although I'm not 100% sure if that's all in the futures_util crate or not)